protocol/src/main/java/org/fido/iot/protocol/epid/EpidMaterialService.java

```diff
@@ -0,0 +1,109 @@
+package org.fido.iot.protocol.epid;
+
+import com.google.common.base.Joiner;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.security.PublicKey;
+import java.util.ArrayList;
+import java.util.List;
+import javax.xml.bind.DatatypeConverter;
+import org.fido.iot.protocol.Composite;
+import org.fido.iot.protocol.Const;
+import org.fido.iot.protocol.CryptoService;
+import org.fido.iot.protocol.InvalidMessageException;
+
+public class EpidMaterialService {
+
+  /**
+   * Combines list of byte array into a single byte array.
+   *
+   * @param data List of byte array containing EPIDresource data
+   * @return byte array of all EPID resources
+   * @throws IOException for unhadled IO exceptions
+   */
+  public static byte[] createLVs(List<byte[]> data) throws IOException {
+
+    return (data.stream()
+        .collect(
+            () -> new ByteArrayOutputStream(),
+            (b, e) -> {
+              try {
+                b.write(e);
+              } catch (IOException ioException) {
+                throw new RuntimeException(ioException);
+              }
+            },
+            (a, b) -> {})
+        .toByteArray());
+  }
+
+  byte[] getSigrl(Composite sigInfo, String epidVersion) {
+    byte[] sigrlResponse = getEpidVerificationServiceResource(sigInfo, Const.SIGRL, epidVersion);
+    if (sigrlResponse == null) {
+      return new byte[] {};
+    }
+    return sigrlResponse;
+  }
+
+  byte[] getPublicKey(Composite sigInfo) {
+    return getEpidVerificationServiceResource(
+        sigInfo, Const.PUBKEY, (Const.EPID_PROTOCOL_VERSION + Const.EPID_20));
+  }
+
+  byte[] getGroupCertSigma10(Composite sigInfo) {
+    return getEpidVerificationServiceResource(
+        sigInfo, Const.GROUPCERTSIGMA10, (Const.EPID_PROTOCOL_VERSION + Const.EPID_11));
+  }
+
+  byte[] getGroupCertSigma11(Composite sigInfo) {
+    return getEpidVerificationServiceResource(
+        sigInfo, Const.GROUPCERTSIGMA11, (Const.EPID_PROTOCOL_VERSION + Const.EPID_11));
+  }
+
+  private byte[] getEpidVerificationServiceResource(
+      Composite sigInfo, String resource, String epidVersion) {
+    String path =
+        Joiner.on(Const.URL_PATH_SEPARATOR)
+            .join(
+                epidVersion,
+                DatatypeConverter.printHexBinary(sigInfo.toBytes()),
+                resource.toLowerCase());
+
+    // todo: Call HTTP Client that will return byte array. Replace return value
+    return null;
+  }
+
+  Composite getSigInfo(Composite sigA, CryptoService cryptoService) throws IOException {
+
+    PublicKey publicKey = cryptoService.decode(sigA.getAsComposite(Const.FIRST_KEY));
```

Comment:
Couple of comments:
1. The switch below at line 86 should be based on the first key of sigA that gives 'sgType', while the second key would provide the GroupID (Section 3.3.3 of spec). Please update lines 78-83 based on this.
2. Additionally, move out the getEpidGroupIdLength() method from cryptoservice to a separate class, something like EpidUtils, since the group id check would be done during the signature verification stage as well. This would eliminate the need of having CryptoService instance as an argument to the mthod.

Reply:
Done.